pkg/controller/mongodb/mongodb_tls_test.go

```diff
@@ -164,12 +158,10 @@ func TestAutomationConfig_IsCorrectlyConfiguredWithTLS(t *testing.T) {
 		for _, process := range ac.Processes {
 			operatorSecretFileName := tlsOperatorSecretFileName("CERT", "KEY")
 
-			assert.Equal(t, automationconfig.MongoDBTLS{
-				Mode:                               automationconfig.TLSModePreferred,
-				PEMKeyFile:                         tlsOperatorSecretMountPath + operatorSecretFileName,
-				CAFile:                             tlsCAMountPath + tlsCACertName,
-				AllowConnectionsWithoutCertificate: true,
-			}, process.Args26.Net.TLS)
+			assert.Equal(t, automationconfig.TLSModePreferred, process.Args26.Get("net.tls.mode").Data())
```

Comment:
I think we should get the values with the `Must*` functions? this one would be MustStr()

Reply:
I considered that but eventually went with `Data` as I thought it would yield better error messages. When using `Must` we can fail either through a panic or because of the assert. If using `Data` we can only fail through the assert and I `testify` will yield a nice error message showing what type we got compared to what type was expected.

I'd be happy to change it though. What do you say?

Reply:
I don't have a strong opinion, but if Data will produce better error messages, it is a good reason to use it.

Reply:
Just did a test and the error messages are definitely better:

```
expected: string("100")
actual  : int(100)
```

vs.

```
panic: interface conversion: interface {} is int, not string [recovered] + stack trace
```